docs/test_scripts/test_custom_model_demo_1.py

```diff
@@ -0,0 +1,146 @@
+
+
+class UNet(nn.Module):
+    def __init__(self, in_channels=1, out_channels=1):
+        super(UNet, self).__init__()
+
+        self.enc1 = self.conv_block(in_channels, 64)
+        self.enc2 = self.conv_block(64, 128)
+
+        self.dec1 = self.upconv_block(128, 64)
+        self.dec2 = self.upconv_block(64, 32)
+        self.final = nn.Conv2d(32, out_channels, kernel_size=1)
+
+    def conv_block(self, in_channels, out_channels):
+        return nn.Sequential(
+            nn.Conv2d(in_channels, out_channels, 3, padding=1),
+            nn.ReLU(inplace=True),
+            nn.MaxPool2d(2),
+        )
+
+    def upconv_block(self, in_channels, out_channels):
+        return nn.Sequential(
+            nn.ConvTranspose2d(in_channels, out_channels, 2, stride=2),
+            nn.Conv2d(out_channels, out_channels, 3, padding=1),
+            nn.ReLU(inplace=True),
+        )
+
+    def forward(self, x):
+        x = self.enc1(x)
+        x = self.enc2(x)
+        x = self.dec1(x)
+        x = self.dec2(x)
+        return self.final(x)
+
+
+# [pytorch model]
+
+# [modulus model]
+
+from dataclasses import dataclass
+from modulus.models.meta import ModelMetaData
+from modulus.models.module import Module
+import torch.nn as nn
+
+
+@dataclass
+class MdlsUNetMetaData(ModelMetaData):
+    name: str = "MdlsUNet"
+    # Optimization
+    jit: bool = False
+    cuda_graphs: bool = True
+    amp_cpu: bool = True
+    amp_gpu: bool = True
+
+
+MdlsUNet = Module.from_torch(UNet, meta=MdlsUNetMetaData)
+
+# [modulus model]
+
+# [modulus sym model]
+
+from modulus.sym.models.arch import Arch
+from modulus.sym.key import Key
+from typing import Optional, Dict
+
+
+class MdlsSymUNet(Arch):
+    def __init__(
+        self,
+        input_keys=[Key("a")],
+        output_keys=[Key("b")],
+        in_channels=1,
+        out_channels=1,
+    ):
+        super(MdlsSymUNet, self).__init__(
+            input_keys=input_keys, output_keys=output_keys
+        )
+
+        self.mdls_model = MdlsUNet(in_channels, out_channels)  # MdlsUNet defined above
+
+    def forward(self, dict_tensor: Dict[str, torch.Tensor]):
+        x = self.concat_input(
+            dict_tensor,
+            self.input_key_dict,
+            detach_dict=None,
+            dim=1,
+        )
+        out = self.mdls_model(x)
+        return self.split_output(out, self.output_key_dict, dim=1)
+
+
+# [modulus sym model]
+
+
+# [code]
+
+import time
+from modulus.utils import StaticCaptureTraining
+
+normaliser = {
+    "permeability": (1.25, 0.75),
+    "darcy": (4.52e-2, 2.79e-2),
+}
+dataloader = modulus.datapipes.benchmarks.darcy.Darcy2D(
+    resolution=256, batch_size=8, nr_permeability_freq=5, normaliser=normaliser
+)
+model = MdlsUNet().to("cuda")
+
+optimizer = torch.optim.Adam(model.parameters(), lr=0.01)
+scheduler = torch.optim.lr_scheduler.LambdaLR(
+    optimizer, lr_lambda=lambda step: 0.85**step
+)
+
+
+# Create training step function with optimization wrapper
```

Comment:
Add comment here saying this also runs the backward on the loss and the optimizer step as well as any mixed precision optimizations and cuda graphs

Reply:
Thanks! Done